src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -73,12 +75,23 @@ private boolean isReady(BlobItem blob, String containerName) {
     }
 
     private void processBlob(BlobClient blob) {
-        envelopeService
-            .findLastEnvelope(blob.getBlobName(), blob.getContainerName())
+        getLastEnvelope(blob)
             .ifPresentOrElse(
                 envelope -> {
                     if (envelope.status == Status.CREATED) {
-                        leaseAndThen(blob, () -> blobProcessor.continueProcessing(envelope.id, blob));
+                        leaseAndThen(blob, () -> {
+                            Optional<Envelope> envelopeOpt = getLastEnvelope(blob);
+                            if (envelopeOpt.isPresent()) {
+                                if (envelopeOpt.get().status == Status.CREATED) {
+                                    blobProcessor.continueProcessing(envelope.id, blob);
+                                } else {
+                                    logger.info(
+                                        "Envelope status changed in system, skipping. {} ",
```

Comment:
Best to have the same message as below, i.e. that the envelope has already been processed in the system. That's the information we care about the most. The fact that its status has changed since we acquired the lease, not so much. We can mention that, though.